src/Data/Nat/Base.agda

```diff
@@ -233,7 +242,40 @@ zero  ! = 1
 suc n ! = suc n * n !
 
 ------------------------------------------------------------------------
--- Alternative definition of _≤_
+-- Alternative definitions of _≤_/_<_
+
+------------------------------------------------------------------------
+-- Another useful (proof-irrelevant) definition of _<_
+
+LessThan : Rel ℕ 0ℓ
+LessThan m n = T (m <ᵇ n)
+
+-- instance
+
+instance
+  lessThan : ∀ {n} → LessThan 0 (suc n)
```

Comment:
Does adding a second instance `{m n} {{LessThan m n}} -> LessThan (suc m) (suc n)` work?

Reply:
Good question, but my most recent commits seem to take a lot longer to typecheck with the instance inference, so i am reluctant to add extra sources of possible useless searching to the mix. But I will see what happens when I do!

Reply:
So... making the instance argument irrelevant fails without patttern matching on `m` and `n`. Making it relevant makes this basically a no-op. not clear what the trade-offs are between the two solutions. What do you advise?

Reply:
But I do wonder what the actual effect of defining
```agda
lessThanSucSuc : ∀ {m n} → ⦃ LessThan m n ⦄ → LessThan (suc m) (suc n)
lessThanSucSuc ⦃ lt ⦄ = lt
```
might be... a no-op, basically, or something more interesting?

Reply:
Compared to 
```agda
lessThanSucSuc {zero}  {suc _} = _
lessThanSucSuc {suc m} {suc n} = lessThanSucSuc {m} {n}
```

But also, I am still actually unclear about the trade-offs involved in writing such functions as 'constructors' of the relation, but using them in local `instance` declarations in subsequent proofs, or else declaring them up-front as `instances` in the module. 

Plus: the warnings about ignoring instance declarations taking explicit arguments seem to stymie one or two subsequent arguments...

Reply:
Now that `LessThan` is defined in terms of `_<ᵇ_`, on further reflection it seems as though `*SucSuc`-like lemmas, or instances, will just be definitional no-ops, so only serve to pollute the name/instance search space... so I think they should *not* be added? let me know if I'm misunderstanding something here?